File: demo-android/app/src/androidTest/java/com/example/demo/e2e/interaction/step/LanguageSwitcherStep.kt
package com.example.demo.e2e.interaction.step

import androidx.compose.ui.test.junit4.AndroidComposeTestRule
import androidx.compose.ui.test.performClick
import com.example.demo.MainActivity
import com.example.demo.e2e.interaction.page.LanguageSwitcherDropdown
import io.qameta.allure.kotlin.Allure

class LanguageSwitcherStep(
    rule: AndroidComposeTestRule<*, MainActivity>,
) {
    private val languageSwitcher = LanguageSwitcherDropdown(rule)

    fun selectLanguage(language: String) {
        Allure.step("Select language '$language'") {
            languageSwitcher.dropdown().performClick()
            when (language) {
                "ES" -> languageSwitcher.spanishOption().performClick()
                "EN" -> languageSwitcher.englishOption().performClick()
                else -> error("Unsupported language: $language")
            }
        }
    }
}

File: demo-ios/DemoUITests/Interaction/Step/LanguageSwitcherStep.swift
import XCTest

final class LanguageSwitcherStep {
    private let languageSwitcher: LanguageSwitcherDropdown

    init(app: XCUIApplication) {
        languageSwitcher = LanguageSwitcherDropdown(app: app)
    }

    func selectLanguage(_ language: String) {
        Allure.step("Select language '\(language)'") {
            languageSwitcher.dropdown().tapWhenReady()
            switch language {
            case "ES":
                languageSwitcher.spanishOption().tapWhenReady(timeout: 5)
            case "EN":
                languageSwitcher.englishOption().tapWhenReady(timeout: 5)
            default:
                XCTFail("Unsupported language: \(language)")
            }
        }
    }
}
